From 0b40548bd9116dc703cc6d8e4ca147c78fa0b41d Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Date: Thu, 26 Jan 2023 12:22:56 +0100
Subject: [PATCH 265/841] fix(st-ddr): stm32mp2: save self-refresh mode after
 each update

Update static variable when calling ddr_set_sr_mode() without any issue.
This variable is used to determine which transition has to be executed
when switching low power mode.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git fc18738845a03c6712efa62152d501867604e261 fix(st-ddr): stm32mp2: save self-refresh mode after each update]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Change-Id: I2d9f3ae11da4d520d280e6621bf78f7af0b2a4cf
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/287295
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 drivers/st/ddr/stm32mp2_ddr_helpers.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/st/ddr/stm32mp2_ddr_helpers.c b/drivers/st/ddr/stm32mp2_ddr_helpers.c
index f2f99fb06..51fd7d41a 100644
--- a/drivers/st/ddr/stm32mp2_ddr_helpers.c
+++ b/drivers/st/ddr/stm32mp2_ddr_helpers.c
@@ -317,6 +317,8 @@ void ddr_set_sr_mode(enum stm32mp2_ddr_sr_mode mode)
 		ERROR("Unknown Self Refresh mode\n");
 		panic();
 	}
+
+	saved_ddr_sr_mode = mode;
 }
 
 void ddr_save_sr_mode(void)
-- 
2.47.2

